src/ui/hooks/settings.ts

```diff
@@ -63,6 +66,20 @@ export function useGetUserSettings() {
   return { userSettings: convertUserSettings(data), error, loading };
 }
 
+export const useFeature = (prefKey: keyof typeof features) => {
+  const [pref, setPref] = useState(features[prefKey]);
+
+  const onUpdate = (prefs: any) => {
+    setPref(prefs.getBoolPref(`devtools.features.${prefKey}`));
+  };
+
+  useEffect(() => {
+    prefsService.addObserver(`devtools.features.${prefKey}`, onUpdate, false);
+  }, []);
```

Comment:
Random review:

I think useEffect in this way is at risk of continuously adding observers as the structure of the page changes, right? At least if the component using this hook were consistently rendered and then removed.

Seems like we'd want more like:
```
export const useFeature = (prefKey: keyof typeof features) => {
  const [pref, setPref] = useState(features[prefKey]);

  const prefName = `devtools.features.${prefKey}`;

  useEffect(() => {
    // Callback inside the hook since it isn't used outside anywhere
    const onUpdate = (prefs: any) => {
      setPref(prefs.getBoolPref(prefName));
    };

    prefsService.addObserver(prefName, onUpdate, false);
    // Include cleanup hook
    return () =>  prefsService.removeObserver(prefName, onUpdate, false);
  }, [
    // Include effect pref name
    prefName,
  ]);

  return pref;
};
```

Reply:
Definitely agree on the remove observer call. I remember thinking I should do that and then clearly I forgot 😆

Reply:
I kind of thought that the cleanup function would get run at the end of the component lifetime, but just looked it up and found out that react will do the cleanup before processing the next call 👍  Thanks for the suggestions @loganfsmyth !